### PR TITLE
Introduce codeql code scanning action

### PR DESCRIPTION
**Pull request by aeisenberg**
Adds a new workflow to run codescanning on every commit.

### CODE REVIEW

**Comment by aeisenberg** [comment]
Closes #402

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+name: "Code Scanning - CodeQL"
+
+on:
```
Comment (jcreedcmu) [inline .github/workflows/codeql.yml:3]: Does this want to be `on: [push, pull_request]` instead?

**Comment by adityasharad** [comment]
`pull_request` is not yet supported and might not work correctly - `push` is what you want (and it will handle PRs created from this repo, though not yet from upstream).

**Comment by jcreedcmu** [comment]
Ok, in that case I merely misdiagnosed the failure of codescanning to immediately run on the PR itself; subsequent PRs seem to be working great with it.

**Comment by aeisenberg** [comment]
@adityasharad so that means in order to run codeql on forks, we need to keep the LGTM integration? If that's the case, I wonder if this code scanning workflow is useful at all.

**Pull request merged**